test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageDirectiveFeatureTest.cs

```diff
@@ -0,0 +1,232 @@
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public class PageDirectiveFeatureTest
+    {
+        [Fact]
+        public void TryGetRouteTemplate_FindsTemplate()
+        {
+            // Arrange
+            string template;
+            var projectItem = new TestRazorProjectItem(@"@page ""Some/Path/{value}""
+The rest of the thing");
+
+            // Act & Assert
+            Assert.True(PageDirectiveFeature.TryGetRouteTemplate(projectItem, out template));
+            Assert.Equal("Some/Path/{value}", template);
+        }
+
+        [Fact]
+        public void TryGetRouteTemplate_NoNewLine()
+        {
+            // Arrange
+            string template;
+            var projectItem = new TestRazorProjectItem(@"@page ""Some/Path/{value}""");
+
+            // Act & Assert
+            Assert.True(PageDirectiveFeature.TryGetRouteTemplate(projectItem, out template));
+            Assert.Equal("Some/Path/{value}", template);
+        }
+        [Fact]
+        public void TryGetRouteTemplate_JunkBeforeDirective()
+        {
+            // Arrange
+            string template;
+            var projectItem = new TestRazorProjectItem(@"Not a directive @page ""Some/Path/{value}""");
+
+            // Act & Assert
+            Assert.False(PageDirectiveFeature.TryGetRouteTemplate(projectItem, out template));
+            Assert.Null(template);
+        }
+
+        [Fact]
+        public void TryGetRouteTemplate_NoQuotesAroundPath()
+        {
+            // Arrange
+            string template;
+            var projectItem = new TestRazorProjectItem(@"@page Some/Path/{value}
+");
+
+            // Act & Assert
+            Assert.True(PageDirectiveFeature.TryGetRouteTemplate(projectItem, out template));
+            Assert.Equal("Some/Path/{value}", template);
+        }
+
+        [Fact]
+        public void TryGetRouteTemplate_WrongNewLine()
+        {
+            // Arrange
+            string wrongNewLine = "/r/n";
+            if (Environment.NewLine == "/r/n")
+            {
+                wrongNewLine = "/n";
+            }
+
+            string template;
+            var projectItem = new TestRazorProjectItem($"@page \"Some/Path/{{value}}\" {wrongNewLine}");
+
+            // Act & Assert
+            Assert.True(PageDirectiveFeature.TryGetRouteTemplate(projectItem, out template));
+            Assert.Equal("Some/Path/{value}", template);
+        }
+
+        [Fact]
+        public void TryGetRouteTemplate_WhitespaceBeforeDirective()
+        {
+            // Arrange
+            string template;
+            var projectItem = new TestRazorProjectItem(@"   @page ""Some/Path/{value}""
+");
+
+            // Act & Assert
+            Assert.True(PageDirectiveFeature.TryGetRouteTemplate(projectItem, out template));
+            Assert.Equal("Some/Path/{value}", template);
+        }
+
+        [Fact(Skip = "Re-evaluate this scenario after we use Razor to parse this stuff")]
```

Comment:
My understanding was that we're not going to worry about this scenario for now, correct me if that's wrong.